components/renku_data_services/storage_models/core.py

```diff
@@ -0,0 +1,31 @@
+"""Models for cloud storage."""
+
+
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass(frozen=True, eq=True, kw_only=True)
+class CloudStorage:
+    """Cloud Storage model."""
+
+    git_url: str
+    storage_type: str
+    configuration: dict[str, Any]
+
+    storage_id: str | None = None
+
+    @classmethod
+    def from_dict(cls, data: dict) -> "CloudStorage":
+        """Create the model from a plain dictionary."""
+        return cls(
+            git_url=data["git_url"],
+            storage_id=data.get("storage_id"),
+            configuration=data["configuration"],
+            storage_type=data.get("storage_type", data["configuration"].get("type")),
```

Comment:
Is this supposed to be the same as the rclone prefix? Is it guaranteed to match it?

Reply:
The validation done using the RClone schema would fail if it's something that doesn't exist in RClone.

We might remove it in the API (and just use `configuration.type` directly, since that has to match anyways), though I want it in the DB at the top level on a separate property so it's easy to query for, to e.g. show "S3" icons in the UI etc.